controllers/users.js

```diff
@@ -496,29 +505,30 @@ class UserController {
    * @returns {Object} Response
    */
   async unfollowUser(req, res) {
-    const { userId: id } = req.params;
-    const { email: following } = req.user;
+    const { userId: following } = req.params;
+    const { email: follower } = req.user;
     const isUserExist = await User.findOne({
       where: {
-        id
+        id: following
       }
     });
     if (!isUserExist) {
       throw new httpError(404, 'User does not exist');
     }
+    const { dataValues: userObj } = isUserExist;
     const isunFollowed = await Follow.findOne({
       where: {
-        id,
-        following
+        follower,
+        following: userObj.email
       }
     });
     if (!isunFollowed) {
-      throw new httpError(404, 'User does not exist');
+      throw new httpError(404, 'User already unfollowed');
     }
-    const dataValues = await Follow.destroy({
+    await Follow.destroy({
```

Comment:
Hi @Jaman-dedy, let's not delete the database content. We may need to use this data for other purposes like in data science. I suggest we have a status which we will be updating instead of deleting it entirely.

Reply:
I worked on this